src/main/java/com/google/devtools/build/lib/bazel/rules/android/AndroidRepositoryFunction.java

```diff
@@ -83,7 +83,7 @@ final DirectoryListingValue getDirectoryListing(Path root, PathFragment dirPath,
   static final ImmutableSortedSet<Integer> getApiLevels(Dirents platformsDirectories) {
     ImmutableSortedSet.Builder<Integer> apiLevels = ImmutableSortedSet.reverseOrder();
     for (Dirent platformDirectory : platformsDirectories) {
-      if (platformDirectory.getType() != Dirent.Type.DIRECTORY) {
+      if (platformDirectory.getType() != Dirent.Type.DIRECTORY && platformDirectory.getType() != Dirent.Type.SYMLINK) {
```

Comment:
So if the platform-* directories are all symlinks on some os's, wouldn't this skip all of the API levels and return nothing? I think android_sdk_repository will throw an error saying that no API levels are installed.

Reply:
Looking at this again, I think I misread the code. This would be clearer I think if we rewrote it to this:

```java
for (Dirent platformDirectory : platformsDirectories) {
  if (platformDirectory.getType() == Dirent.Type.DIRECTORY || platformDirectory.getType() == Dirent.Type.SYMLINK) {
    Matcher matcher = PLATFORMS_API_LEVEL_PATTERN.matcher(platformDirectory.getName());
    if (matcher.matches()) {
      apiLevels.add(Integer.parseInt(matcher.group(1)));
    }
  }
}
```

Reply:
fixed :+1: